sw/device/lib/dif/dif_kmac.c

```diff
@@ -4,4 +4,55 @@
 
 #include "sw/device/lib/dif/dif_kmac.h"
 
+#include "sw/device/lib/base/memory.h"
+
 // TODO: implement!
+
+dif_kmac_result_t dif_kmac_customization_string_init(
+    const char *data, size_t len, dif_kmac_customization_string_t *out) {
+  if ((data == NULL && len != 0) || out == NULL) {
+    return kDifKmacBadArg;
+  }
+
+  if (len > kDifKmacMaxCustomizationStringLen) {
+    return kDifKmacBadArg;
+  }
+
+  // Convert length in bytes to length in bits.
+  uint16_t bits = (uint16_t)(len)*8;
```

Comment:
If you want to not run awry of promo rules I'd write this as `((uint16_t) len) * 8`.

Reply:
This looks like something I didn't know, so I'd love to hear more. Doesn't the cast bind tighter than the multiplication, so the two options parse to the same expression?

Reply:
I've changed it. In this case the cast is just to document that a conversion is happening here, it doesn't have any actual effect.

Reply:
Rupert: that's correct, but we prefer not to rely on that because most people rarely remember the subtleties when bit operations become involved. =)

There's a reason no programming language in the last decade has promotion rules.

Reply:
Oh, right. I think you're talking about operator precedence, not promotion rules. But, yep, makes sense.